### PR TITLE
make a few changes listed below

### PR DESCRIPTION
**Pull request by begoat**
1. ignore vscode file
2. add OS X travis-ci support
3. roll back the main.cpp to initial status
4. add a functionality that can support you refresh you app after it starts when changes are made in qml file.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,5 @@ GameHall_autogen
 .AppleDouble
 .LSOverride
 
+# Vscode
+.vscode
```
Comment (sunnyadn) [inline .gitignore]: Actually the line CANNOT work. Follow the below link as an example to ignore user config file of VSCode

https://github.com/github/gitignore/blob/master/Global/VisualStudioCode.gitignore

File: CMakeLists.txt [hunk 2]
```diff
@@ -2,12 +2,20 @@ cmake_minimum_required(VERSION 2.8.12)
 
 project(GameHall LANGUAGES CXX)
 
+set(Watch_Dir_Path "\"${PROJECT_SOURCE_DIR}/script/QML\"")
```
Comment (sunnyadn) [inline CMakeLists.txt]: **ALWAYS** USE **UPPER CASE** AS A MACRO NAME.  😄

File: src/WatchAndReload/WatchReload.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        this->contentObj->setProperty("active",false);
+
+        qDebug() << "1111";
+
+        this->engine->clearComponentCache();
+        this->headerObj->setProperty("source",headerPath);
+        this->contentObj->setProperty("source",contentPath);
+        this->engine->clearComponentCache();
+
+        qDebug() << "1111";
```
Comment (sunnyadn) [inline src/WatchAndReload/WatchReload.cpp]: AVOID using meaningless debug info!

File: src/WatchAndReload/WatchReload.h [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#ifndef WatchReload_H
+#define WatchReload_H
+#include <QQmlApplicationEngine>
+
+class WatchReload: public QObject
+{
+    Q_OBJECT
+public:
+  WatchReload(QQmlApplicationEngine *engine);
```
Comment (sunnyadn) [inline src/WatchAndReload/WatchReload.h]: 1 indent = **4** spaces

**Review by sunnyadn — CHANGES_REQUESTED** [review]

**Comment by sunnyadn** [comment]
https://github.com/Mogara/QSanguosha-For-Hegemony/wiki/Coding-Style
Your coding style is a bit far from that of Qt. Please read this doc as a reference.

**Comment by begoat** [comment]
okay, i will refer to the link and do some changes.

**Review by sunnyadn — APPROVED** [review]

**Pull request merged**